interview-questions.md

```diff
@@ -7,44 +8,58 @@ Then, research the question to expand on your answer. Even if you feel you have
+Researched answer: Post, Patch, Delete, Get. The reason params are passed so the Rails router recognizes URLs and dispatches them to a controller’s action. It can also generate paths and URLs, avoiding the need to hardcode strings in your views.
 
-  Your answer:
+3. Name three rails generator commands. What is created by each?
 
-  Researched answer:
+Your answer: rails s - runs the server, rails db:create - sets up a database
 
+Researched answer: rails new myapp -d postgresql -T, rails s - runs the server, rails db:create - sets up a database.
 
+4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
 
-3. Name three rails generator commands. What is created by each?
+method="GET" /students shows a list of the students
 
-  Your answer:
+method="GET" /students/2 shows a specfic student
 
-  Researched answer:
+method="GET" /students/new returns HTML form from the student
 
+method="GET" /students/edit/2 edits a student and returns an HTML form
 
+method="POST" /students/ The post sends the student to the DB
 
-4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
+method="PATCH" /students/2 THe patch is used to update -student
 
-method="GET"    /students          
+method="DELETE" /students/2 Delete will destroy what ID number you want to delete(student)
 
-method="GET"    /students/2  
+5. As a developer, you want to make an application that will help you manage your to do list. To get started, you want to create the API. Create 10 user stories that will help you get your application started.
 
-method="GET"    /students/new
+Creating an API story using Cars.
 
-method="GET"    /students/edit/2    
+As a developer I was to create a to do (list) in the DB. The to do list would have the following info: Tasks, supplies, time.
 
-method="POST"   /students/       
+As a developer when i put Lists.all I should see a list of all my lists in the Db.
 
-method="PATCH"  /students/2      
+As a developer I want to see errors if im unable to add a task to the to do list.
 
-method="DELETE" /students/2      
+I want to see errors if my to do list doesn't update once a task is complete.
 
+I want to see validation errors if the to do list starts to repeat a task
 
+I want to see errors if i cant add a list to the time.
 
-5. As a developer, you want to make an application that will help you manage your to do list. To get started, you want to create the API. Create 10 user stories that will help you get your application started.
+I want to see errors if my list cant update the supplies.
+
+I want to see errors if my list can add a task to the list.
+
+As a developer I want to be able to retrieve the supplies information if I wanted to enhance the API.
+
+I want to see errors if my to do list cant remove a task when deleted.
```

Comment:
Kinda, User stories are hard to get used to but very helpful
"As a User I have access to a database called ToDoList"
"As a user I can create a new task in the ToDoList"